src/aktualizr_secondary/aktualizr_secondary_factory.h

```diff
@@ -0,0 +1,13 @@
+#ifndef AKTUALIZR_SECONDARY_FACTORY_H
+#define AKTUALIZR_SECONDARY_FACTORY_H
+
+#include "aktualizr_secondary.h"
+
+class AktualizrSecondaryFactory {
+ public:
+  static AktualizrSecondary::Ptr create(const AktualizrSecondaryConfig& config);
+  static AktualizrSecondary::Ptr create(const AktualizrSecondaryConfig& config,
+                                        const std::shared_ptr<INvStorage>& storage);
```

Comment:
To avoid code duplication (many clients doing the same prep work before calling AktualizrSecondary ctor);
To decouple AktualizrSecondary from its ctor parameters specialization(concrete classes), it should depend just on interfaces and/or base classes;

Reply:
Okay, I think you've convinced me here. `Factory` can be a scary word but I don't think it's being abused or misused here. :)